src/main/java/org/broadinstitute/hellbender/tools/walkers/annotator/VariantAnnotatorEngine.java

```diff
@@ -0,0 +1,296 @@
+    }
+
+    private VariantOverlapAnnotator initializeDBs(final FeatureInput<VariantContext> dbSNPInput) {
+        final Map<FeatureInput<VariantContext>, String> overlaps = new LinkedHashMap<>();
+        if ( dbSNPInput != null ) {
+            overlaps.put(dbSNPInput, VCFConstants.DBSNP_KEY); // add overlap detection with DBSNP by default
+        }
+
+        return new VariantOverlapAnnotator(dbSNPInput, overlaps);
+    }
+
+    /**
+     * Returns the list of genotype annotations that will be applied.
+     * Note: The returned list is unmodifiable.
+     */
+    public List<GenotypeAnnotation> getGenotypeAnnotations() {
+        return Collections.unmodifiableList(genotypeAnnotations);
+    }
+
+    /**
+     * Returns the list of info annotations that will be applied.
+     * Note: The returned list is unmodifiable.
+     */
+    public List<InfoFieldAnnotation> getInfoAnnotations() {
+        return Collections.unmodifiableList(infoAnnotations);
+    }
+
+    /**
+     * Returns the set of descriptions to be added to the VCFHeader line (for all annotations in this engine).
+     */
+    public Set<VCFHeaderLine> getVCFAnnotationDescriptions() {
+        final Set<VCFHeaderLine> descriptions = new HashSet<>();
+
+        for ( final InfoFieldAnnotation annotation : infoAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final GenotypeAnnotation annotation : genotypeAnnotations) {
+            descriptions.addAll(annotation.getDescriptions());
+        }
+        for ( final String db : variantOverlapAnnotator.getOverlapNames() ) {
+            if ( VCFStandardHeaderLines.getInfoLine(db, false) != null ) {
+                descriptions.add(VCFStandardHeaderLines.getInfoLine(db));
+            } else {
+                descriptions.add(new VCFInfoHeaderLine(db, 0, VCFHeaderLineType.Flag, db + " Membership"));
+            }
+        }
+
+        return descriptions;
+    }
+
+    /**
+     * Annotates the given variant context - adds all annotations that satisfy the predicate.
+     * @param features context containing the features that overlap the given variant
+     * @param ref the reference context of the variant to annotate or null if there is none
+     * @param vc the variant context to annotate
+     * @param stratifiedPerReadAlleleLikelihoodMap par-sample map of likelihoods per read. May be null
+     * @param addAnnot function that indicates if the given annotation type should be added to the variant
+     */
+    public VariantContext annotateContext(final FeatureContext features,
+                                          final ReferenceContext ref,
+                                          final VariantContext vc,
+                                          final Map<String, PerReadAlleleLikelihoodMap> stratifiedPerReadAlleleLikelihoodMap,
+                                          final Predicate<VariantAnnotation> addAnnot) {
```

Comment:
It would be more logical for `vc` to be the first method parameter here, since that is the object being annotated. The current ordering, with `features` first, implies that `features` is being annotated, which is not the case.